### PR TITLE
Removed legacy name column and views related to it

### LINKED ISSUE #323: Move legacy URLs to Nginx config
Funnelapp has some code to handle legacy `funnel.hasgeek.com/<project>` URLs (at the bottom of `views/index.py`). This code should be removed and redirects for these projects should be moved to Nginx config. The `legacy_name` attribute on the `Project` model should be dropped as well.

### PR DESCRIPTION
**Pull request by iambibhas**
Fixes #323.

### CODE REVIEW

File: migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def downgrade():
+    conn = op.get_bind()
+
+    op.add_column('project', sa.Column('legacy_name', sa.Unicode(length=250), nullable=True))
+    op.create_unique_constraint('project_legacy_name_key', 'project', ['legacy_name'])
+
+    for data in legacy_data:
+        prof = conn.execute(sa.select([profile.c.id]).where(profile.c.name == data['profile_name']).limit(1)).fetchone()
```
Comment (jace) [inline migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py]: Will this downgrade fail on a database that does not have these entries?

File: migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    {'legacy_name': u'fifthel2014', 'name': u'2014', 'profile_name': u'fifthelephant'},
+    {'legacy_name': u'droidcon2014', 'name': u'2014', 'profile_name': u'droidconin'},
+    {'legacy_name': u'jsfoo2014', 'name': u'2014', 'profile_name': u'jsfoo'},
+    {'legacy_name': u'metarefresh2015', 'name': u'2015', 'profile_name': u'metarefresh'},
+    {'legacy_name': u'rootconf2014', 'name': u'2014', 'profile_name': u'rootconf'},
+    {'legacy_name': u'metarefresh2014', 'name': u'2014', 'profile_name': u'metarefresh'},
+    {'legacy_name': u'angularjs-miniconf-2014', 'name': u'2014-angularjs', 'profile_name': u'miniconf'},
+    {'legacy_name': u'droidcon2013', 'name': u'2013', 'profile_name': u'droidconin'},
+    {'legacy_name': u'redis-miniconf-2014', 'name': u'2014-redis', 'profile_name': u'miniconf'},
+    {'legacy_name': u'rootconf-miniconf-2014', 'name': u'2014-rootconf', 'profile_name': u'miniconf'}]
```
Comment (jace) [inline migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py]: Include the nginx config as a text file `legacy_redirects.txt`.

File: migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py (hunk 1 above)
Comment (iambibhas) [reply migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py]: It'll fail if the database does not have the given profile.

Comment (iambibhas) [reply migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py]: And the projects.

Comment (jace) [reply migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py]: Add safeguards then.

File: migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py [hunk 3]
```diff
@@ -71,4 +71,5 @@ def downgrade():
 
     for data in legacy_data:
         prof = conn.execute(sa.select([profile.c.id]).where(profile.c.name == data['profile_name']).limit(1)).fetchone()
```
Comment (jace) [inline migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py]: I wish you had called this `profile`. I had to scroll up and down to figure out why you were referring to a professor.

File: migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py (hunk 2 above)
Comment (iambibhas) [reply migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py]: Done.

File: migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py (hunk 3 above)
Comment (iambibhas) [reply migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py]: 😓

Comment (iambibhas) [reply migrations/versions/4b7fe9b25c6c_removed_legacy_name_from_project.py]: `profile` is already the table name. I'll rename it to something verbose.

**Pull request merged**